Articles/Blog/HelloHeatEquation.md

```diff
@@ -0,0 +1,226 @@
+
+_u_<sup>_k_+1</sup><sub>_i_</sub> = _ru_<sup>_k_</sup><sub>_i_+1</sub> + (1 - 2 _r_)_u_<sup>_k_</sup><sub>_i_</sub> + _ru_<sup>_k_</sup><sub>_i_-1</sub>
+(The full discretization process can be found [here](https://xsdk-project.github.io/ATPESC2018HandsOnLessons/lessons/hand_coded_heat/ "Hand-Coded Heat").)
+
+Where _r = &alpha;_ &Delta;_t&sup2; /_ &Delta;_x&sup2;_. This is the Forward-Time, Central-Space (FTCS) scheme which is considered numerically stable as long as _r_ is less than or equal to 0.5.
+
+### Why the Heat Equation?
+PDEs are widely used among the different sciences to model various phenomena such as electromagnetism, fluid mechanics, and heat transfers. While PDEs extend across multiple fields, they can still be quite tricky to work with because unlike ordinary differential equations, they are comprised of derivatives of more than just a single variable. A single example of a PDE is the Heat Equation, which is used calculate the distribution of heat on a region over time. While here we just focus on the 1-dimensional version of the Heat Equation, it can actually take a multitude of forms including the [Fourier](https://fenix.tecnico.ulisboa.pt/downloadFile/3779571609326/transp3.pdf "Fourier Equation"), [LaPlace](https://www.britannica.com/science/Laplaces-equation "LaPlace's Equation") (also known as steady-state), [2D](http://ramanujan.math.trinity.edu/rdaileda/teach/s12/m3357/lectures/lecture_3_6_short.pdf "2D Heat Equation"), and [3D](https://ocw.mit.edu/courses/mathematics/18-303-linear-partial-differential-equations-fall-2006/lecture-notes/pde3d.pdf "3D Heat Equation") heat equations.
+
+By taking a continuous PDE and turning it into a discrete formula, parallelism can be accomplished. For example, the equation for finding the tempurature at a given point in time in the 1-dimensional Heat Equation is:
+
+_u_<sup>_k_+1</sup><sub>_i_</sub> = _ru_<sup>_k_</sup><sub>_i_+1</sub> + (1 - 2 _r_)_u_<sup>_k_</sup><sub>_i_</sub> + _ru_<sup>_k_</sup><sub>_i_-1</sub>
+
+with _k_ being the time, and _i_ being a point in our axis. We can see that the temperature of _u_<sup>_k+1_</sup><sub>_i_</sub> is only dependent on temperatures from spots _i_, _i-1_, and _i+i_ from the previous iteration. By having two arrays to store of old and new temperatures, each location for time _k_ is independent and can be distributed among different processors and cores.
+
+As a consequence of PDEs being so expansive, several implementations are available for multiple programming languages and numerical libraries. Users who are new to coding are often first introduced to the Heat Equation as a means of teaching them different programming languages or tools such as MPI and OpenMP. By reviewing these implementations a solid foundation of specfic language's characteristics can be seen.
+
+## Implementations
```

Comment:
I *love* all the implementations here. However, for the blog post. I would just list them with links to the actual code.

Reply:
We chose to leave only a few examples in. Do you think removing all the code makes sense? Or, just reducing the examples even more? I feel like it's difficult to talk about the differences in the examples without showing them directly.

Reply:
Remove all the code except one; mabe the C. List the implementation names and make the names (perma-)links to them in the repo.